src/server/JasmineGraphInstanceService.cpp

```diff
@@ -982,167 +975,190 @@ bool JasmineGraphInstanceService::duplicateCentralStore(int thisWorkerPort, int
+        if (response.compare(JasmineGraphInstanceProtocol::SEND_FILE_LEN) != 0) {
+            instance_logger.error("Incorrect response. Expected: " + JasmineGraphInstanceProtocol::SEND_FILE_LEN +
+                                  " ; Received: " + response);
+            close(sockfd);
+            continue;
+        }
+        instance_logger.info("Received : " + JasmineGraphInstanceProtocol::SEND_FILE_LEN);
+        if (Utils::send_str_wrapper(sockfd, fileLength)) {
+            instance_logger.info("Sent : File length in bytes " + fileLength);
+        }
 
-                            response = Utils::read_str_wrapper(sockfd, data, INSTANCE_DATA_LENGTH, false);
-                            if (response.compare(JasmineGraphInstanceProtocol::SEND_FILE_LEN) == 0) {
-                                instance_logger.log("Received : " + JasmineGraphInstanceProtocol::SEND_FILE_LEN,
-                                                    "info");
-                                if (Utils::send_str_wrapper(sockfd, fileLength)) {
-                                    instance_logger.log("Sent : File length in bytes " + fileLength, "info");
-                                }
-
-                                response = Utils::read_str_wrapper(sockfd, data, INSTANCE_DATA_LENGTH, false);
-                                if (response.compare(JasmineGraphInstanceProtocol::SEND_FILE_CONT) == 0) {
-                                    instance_logger.log("Received : " + JasmineGraphInstanceProtocol::SEND_FILE_CONT,
-                                                        "info");
-                                    instance_logger.log("Going to send file through service", "info");
-                                    JasmineGraphInstanceService::sendFileThroughService(host, dataPort, fileName,
-                                                                                        centralStoreFile, masterIP);
-                                }
-                            }
-                        }
-                        int count = 0;
+        response = Utils::read_str_wrapper(sockfd, data, INSTANCE_DATA_LENGTH, false);
+        if (response.compare(JasmineGraphInstanceProtocol::SEND_FILE_CONT) != 0) {
+            instance_logger.error("Incorrect response. Expected: " + JasmineGraphInstanceProtocol::SEND_FILE_CONT +
+                                  " ; Received: " + response);
+            close(sockfd);
+            continue;
+        }
+        instance_logger.info("Received : " + JasmineGraphInstanceProtocol::SEND_FILE_CONT);
 
-                        while (true) {
-                            if (Utils::send_str_wrapper(sockfd, JasmineGraphInstanceProtocol::FILE_RECV_CHK)) {
-                                instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::FILE_RECV_CHK, "info");
-                            }
+        instance_logger.info("Going to send file through service");
```

Comment:
Can we improve this message? In its current form it does not have any meaning. A better message would be `Going to send central store file through file transfer service from worker to master`

Reply:
It's from master to worker. Included the change in next PR.

Reply:
Ok